### ISSUE TITLE
meterplugs perception - gui no working

### ISSUE THREAD

**Issue by lentferj** [falkTX/Carla#504]
I have a problem running meterplugs perception in win32-bridge mode. The plugins (there are 2 actually, source and controller) do work properly, but I can't get the GUI to show, so I can't >actually< use them.

When I open the GUI from carla, I only get tiny windows with nothing in it and in the logs it says:
- WARNING 23:52:03: get_short_path: Failed to convert to short path (2).
- WARNING 23:52:03: Plugin: Failed to get module file name (MeterPlugs Perception Controller.dll).
- WARNING 23:52:03: Plugin: Failed to get module directory.

I would think this is a problem of Carla accessing the files in the wine installation path. The dll files seem to be not "self-contained" but store stuff in subdirectories:

$ pwd
/home/lentferj/.wine/drive_c/Program Files/Steinberg/VSTPlugins
lentferj@ts-d20:~/.wine/drive_c/Program Files/Steinberg/VSTPlugins$ ls -l
insgesamt 8108
drwxr-xr-x 2 lentferj lentferj     137 Jul 22 10:17 MeterPlugs Perception Controller Data
-rw-r--r-- 1 lentferj lentferj 3808256 Dez 13  2016 MeterPlugs Perception Controller.dll
drwxr-xr-x 2 lentferj lentferj     119 Jul 22 10:17 MeterPlugs Perception Source Data
-rw-r--r-- 1 lentferj lentferj 3757568 Dez 13  2016 MeterPlugs Perception Source.dll
-rw-r--r-- 1 lentferj lentferj    5874 Jul 22 10:17 unins000.dat
-rwxr-xr-x 1 lentferj lentferj  710872 Jul 22 10:17 unins000.exe
-rw-r--r-- 1 lentferj lentferj   11225 Jul 22 10:17 unins000.msg

$ pwd
/home/lentferj/.wine/drive_c/Program Files/Steinberg/VSTPlugins/MeterPlugs Perception Controller Data
lentferj@ts-d20:~/.wine/drive_c/Program Files/Steinberg/VSTPlugins/MeterPlugs Perception Controller Data$ ls -l
insgesamt 412
-rw-r--r-- 1 lentferj lentferj   1876 Nov 16  2016 authorize.lua
-rw-r--r-- 1 lentferj lentferj    962 Okt 19  2016 authorize.xml
-rw-r--r-- 1 lentferj lentferj    709 Apr 13  2014 balance.xml
-rw-r--r-- 1 lentferj lentferj  12314 Feb 29  2016 interface.lua
-rw-r--r-- 1 lentferj lentferj 370479 Mai 21  2014 interface.pk3
-rw-r--r-- 1 lentferj lentferj  14260 Feb 29  2016 interface.xml
-rw-r--r-- 1 lentferj lentferj   2758 Feb 29  2016 ui.lua

I have a similar problem with some Melda Plugins and ImpOSCAR 2 - the plugins work but I cannot access the preset files.
Any way to get this to work? TIA, Jan

**Comment by falkTX** [comment]
Carla does not support VST presets yet, so it's normal that it doesn't work yet.

Perception Controller/Source does not have a demo available, so I can't try this.

**Comment by lentferj** [comment]
Yeah, I know. I will try if their two plugins which have demos show the same problem and come back here. Anything else I can do to help debugging? My gut feeling is that when run from carla the plugin just doesn't find it's files in the wine path. It works without any issues when run in airwave.

**Comment by lentferj** [comment]
I installed their LCAST Demo (http://www.meterplugs.com/download) and it shows the same "symptoms".

**Comment by falkTX** [comment]
I tried it, seems to me this is an issue with wine.
Try it in other hosts, still on linux, and see if it works there.

**Comment by lentferj** [comment]
Oh, I should have mentioned that before :/. I works in Bitwig, no problem. It does open in Ardour (via Airwave) and the GUI is visible, but Ardour crashed after a few seconds.

**Comment by lentferj** [comment]
I got it to run in Ardour 5.10 now using LinVST. Funnily I had the same issue in the first place as I described for carla. I did put the Windows dll in ~/.vst_linvst and rename the linvst.so accordingly. I only got an empty plugin window when opening the plugin gui in Ardour. I also hat to copy the "MeterPlugs Perception Controller Data" and "MeterPlugs Perception Source Data" to the path where the dlls resied. This fortifies my thought that carla maybe just not be "finding" this directory. When I load the plugin in carla it's loaded from its original ~/.wine path...

**Comment by lentferj** [comment]
So now I am quite sure it's a problem with the access to the Data directories within the wine prefix. If I build a directory structure like this, it works.

$ pwd
/home/lentferj/.vst_test
lentferj@ts-d20:~/.vst_test$ ls -al
insgesamt 7408
drwxr-xr-x   4 lentferj lentferj    4096 Aug 16 18:31 .
drwxr-xr-x 139 lentferj lentferj    8192 Aug 16 18:30 ..
drwxr-xr-x   2 lentferj lentferj     137 Aug 16 18:31 MeterPlugs Perception Controller Data
-rw-r--r--   1 lentferj lentferj 3808256 Aug 16 18:31 MeterPlugs Perception Controller.dll
drwxr-xr-x   2 lentferj lentferj     119 Aug 16 18:31 MeterPlugs Perception Source Data
-rw-r--r--   1 lentferj lentferj 3757568 Aug 16 18:31 MeterPlugs Perception Source.dll
